fe/fe-core/src/main/java/com/starrocks/load/pipe/Pipe.java

```diff
@@ -0,0 +1,501 @@
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//   http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package com.starrocks.load.pipe;
+
+import com.google.common.base.Preconditions;
+import com.google.gson.annotations.SerializedName;
+import com.starrocks.analysis.TableName;
+import com.starrocks.catalog.Database;
+import com.starrocks.catalog.TableFunctionTable;
+import com.starrocks.common.DdlException;
+import com.starrocks.common.ErrorCode;
+import com.starrocks.common.ErrorReport;
+import com.starrocks.common.Pair;
+import com.starrocks.common.UserException;
+import com.starrocks.persist.gson.GsonPostProcessable;
+import com.starrocks.persist.gson.GsonUtils;
+import com.starrocks.scheduler.Constants;
+import com.starrocks.scheduler.ExecuteOption;
+import com.starrocks.scheduler.SubmitResult;
+import com.starrocks.scheduler.Task;
+import com.starrocks.scheduler.TaskBuilder;
+import com.starrocks.scheduler.TaskManager;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.sql.ast.pipe.CreatePipeStmt;
+import com.starrocks.sql.ast.pipe.PipeName;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.collections.MapUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+/**
+ * Pipe: continuously load and unload data
+ */
+public class Pipe implements GsonPostProcessable {
+
+    private static final Logger LOG = LogManager.getLogger(Pipe.class);
+    public static final int FAILED_TASK_THRESHOLD = 5;
+
+    @SerializedName(value = "name")
+    private final String name;
+    @SerializedName(value = "id")
+    private final PipeId id;
+    @SerializedName(value = "type")
+    private final Type type;
+    @SerializedName(value = "state")
+    private State state;
+    @SerializedName(value = "originSql")
+    private String originSql;
+    @SerializedName(value = "pipeSource")
+    private PipeSource pipeSource;
+    @SerializedName(value = "targetTable")
+    private TableName targetTable;
+    @SerializedName(value = "properties")
+    private Map<String, String> properties;
+    @SerializedName(value = "load_status")
+    private LoadStatus loadStatus = new LoadStatus();
+
+    private ReentrantReadWriteLock lock = new ReentrantReadWriteLock();
+    private Map<Long, PipeTaskDesc> runningTasks = new HashMap<>();
```

Comment:
Will the running tasks be replayed when FE restarts?

Reply:
This part has not been fully implemented, including persistent FileList, and FileList recovery.